client/state/comments/selectors.js

```diff
@@ -108,15 +123,27 @@ export const getPostCommentsTree = createSelector(
 				? filter( allItems, item => item.isPlaceholder || item.status === status )
 				: allItems;
 
+		const [ roots, children ] = partition( items, item => item.parent === false );
+		const parentToChildMap = mapValues( groupBy( children, 'parent.ID' ), mapping =>
+			map( mapping, 'ID' )
+		);
+
+		const transformItemToNode = item => ( {
+			data: item,
+			children: ( parentToChildMap[ item.ID ] || [] ).reverse(),
+		} );
+
+		const rootNodes = map( roots, transformItemToNode );
```

Comment:
i may be misunderstanding the next 5 lines, but is there any logic taking advantage of the child/root split? at first naive glance it looks to me like you could write
```js
const nodes = keyBy(
  map( items, transformItemToNode ),
  'data.ID',
)
```

Reply:
this was late at night, so ... yeah. can likely consolidate.